commands/chain.go

```diff
@@ -122,3 +125,62 @@ var storeStatusCmd = &cmds.Command{
 		return nil
 	},
 }
+
+var storeSetHeadCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Modify the head of the current chain.",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cids", true, true, "CID's of the tipset to set the chain head to."),
+	},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		headCids := make([]cid.Cid, len(req.Arguments))
+		for i, arg := range req.Arguments {
+			c, err := cid.Decode(arg)
+			if err != nil {
+				return err
+			}
+			headCids[i] = c
+		}
+		headKey := types.NewTipSetKey(headCids...)
+
+		headTs, err := GetPorcelainAPI(env).ChainTipSet(headKey)
+		if err != nil {
+			return err
+		}
+		return GetPorcelainAPI(env).ChainSetHead(req.Context, headTs)
+	},
+}
+
+var storeSyncCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Instruct the chain syncer to sync a specific chain head, going to network if required.",
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("peerid", true, false, "Base58-encoded libp2p peer ID to sync from"),
+		cmdkit.StringArg("cids", true, true, "CID's of the tipset to set the chain head to."),
+	},
+	Options: []cmdkit.Option{},
+	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
+		syncPid, err := peer.IDB58Decode(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+
+		syncCids := make([]cid.Cid, len(req.Arguments[1:]))
+		for i, arg := range req.Arguments[1:] {
+			c, err := cid.Decode(arg)
+			if err != nil {
+				return err
+			}
+			syncCids[i] = c
+		}
+		syncKey := types.NewTipSetKey(syncCids...)
+		ci := &types.ChainInfo{
+			Peer:   syncPid,
+			Height: 0, // because this is only checked when trusted is true.
```

Comment:
But trusted *is* set to true below.... I think you meant "is not checked".

Reply:
Opps, thanks.